### DISCUSSION TITLE
Adding VehicleInVehicle cargo areas to vehicles that are missing it

### DISCUSSION THREAD

**Discussion by dedmen** [acemod/ACE3#8606] [Ideas]
I'm just making this in order to coordinate and keep track of this.

Now that we have ACE cargo using vanilla vehicle in vehicle cargo with #7984
Should we add cargo support to some vehicles that are missing?

For example the back of the quad could fit two ammoboxes (If there is no second passenger, with a second passenger its completely messed up, not sure how/if we can handle that? I think we can lock the second passenger seat via script so that noone can get into it if boxes are loaded?)
Or the Stomper can fit some too, we could add engine cargo areas to vehicles.
I put two large plastic boxes onto a stomper yesterday, and it was awesome that people could just take stuff out of the boxes directly instead of first unloading from ACE cargo
![](https://i.imgur.com/eLiggDW.png)

Maybe we could support that by just adding a ViV cargo area to the vehicle.
My main concern is people building missions with ACE and using that, and then running them without ACE and it not working right.
But I think we already don't support that

Arma 2.06 doesn't support adding cargo areas because the config only supports specifying memory points.
But I will add support for this in Arma 2.08 (Q1 2022)
so we could already think about whether we want to do this, and where to do this.

My first ideas would be as I said above Stomper and Quad, having boxes you transport with the Stomper be on top of it, and being able to directly access them is a really nice feature.

This really is a cool feature if supported, and if we get the ability to add support for this to more vehicles that will be great

https://user-images.githubusercontent.com/3768165/138597557-4d8e5c67-843d-462e-9d40-3a47ad9cc8d4.mp4

I have created a Steam Composition and some helper scripts to help with finding the dimensions for Cargo Areas:
https://steamcommunity.com/sharedfiles/filedetails/?id=2636146909

So we could already prepare some code, though all of RHS already seems to have good cargo areas, and vanilla there are only a handful vehicle that are missing it.

Config will be
```
cargoBayDimensions[]= {
					{x,y,z},
					{x,y,z}"
				};
```
This change will only be semi backwards compatible.
If you add this into config and run it in 2.06, you will get config reading error messages in RPT.
probably something like `'cargoBayDimensions/' is not a string` because engine tries to read it as string currently.
Though if you are willing to ignore the errors then the config entries could already be added now.

If you take #7984 and #8604 and this, you can do this:
![20211024173711_1](https://user-images.githubusercontent.com/3768165/138601379-05fad47d-f851-40e3-b3d4-b207537194ee.jpg)
Although the rope bugs around quite alot here, that can probably be solved with custom ropes for the fuel hose, just making it a bit stiffer.

**Comment by dedmen** [comment]
I also just made myself a ticket to fix that cargo boxes cannot fit rotated.
Here is an example of the RHS MRZR which can fit 12 jerrycans in its cargo, but it cannot fit even the smallest plastic box, simply because it cannot rotate by 90°
(Left is placed manually in Eden to show how it would look like)
![20211024161704_1](https://user-images.githubusercontent.com/3768165/138598241-b89cfd27-1254-4d5e-af98-810b451c69ff.jpg)
I will try to fix that for 2.08 too to make this even more useful.

**Reply by Vdauphin** [reply]
> but it cannot fit even the smallest plastic box, simply because it cannot rotate by 90°

Damn, this would be a crazy change for ViV

**Reply by dedmen** [reply]
I think I will just do a stupid
> Doesn't fit?
> Try again but 90° rotated

I think that would already be sufficient.
Could also add a config entry in the to-be-loaded vehicles like these boxes there to specify the default orientation when in cargo.
Will have to see, maybe I can get it next month otherwise we'll see it in January probably

**Comment by tbeswick96** [comment]
```
Arma 2.06 doesn't support adding cargo areas because the config only supports specifying memory points.
But I will add support for this in Arma 2.08 (Q1 2022)
```
This alone will be massive, can't wait!

**Comment by ZluskeN** [comment]
I did some experimenting with this and here are some config values to add VIV for vanilla UGV and transport trucks:

```
class CfgVehicles 
{
	class Car_F;
	class UGV_01_base_F: Car_F {
		class VehicleTransport
		{
			class Carrier
			{
				cargoAlignment[] = { "right", "front" };
				cargoBayDimensions[] = { {-0.15,0.35,-1}, {1.05,-0.85,0.6} };
				disableHeightLimit = 0;
				exits[] = { {5,0,0}, {5,10,0} };
				loadingAngle = 60;
				loadingDistance = 5;
				maxLoadMass	= 3000;
				unloadingInterval = 2;
				cargoSpacing[] = { 0, 0.15, 0 };
			};
		};
	};
	class UGV_01_rcws_base_F: UGV_01_base_F {
		class VehicleTransport
		{
			class Carrier
			{
				cargoAlignment[] = { "right", "front" };
				cargoBayDimensions[] = { {-0.15,0.35,-1}, {1.05,-0.45,-0.2} };
				disableHeightLimit = 0;
				exits[] = { {5,0,0}, {5,10,0} };
				loadingAngle = 60; (in de
				loadingDistance = 5;
				maxLoadMass = 2500;
				unloadingInterval = 2;
				cargoSpacing[] = { 0, 0.15, 0 };
			};
		};
	};
	class Truck_01_base_F;
	class B_Truck_01_transport_F: Truck_01_base_F {
		class VehicleTransport
		{
			class Carrier
			{
				cargoAlignment[] = { "front", "center" };
				cargoBayDimensions[] = { {-0.35,0.65,-0.55}, {0.45,-4.95,0.45} };
				disableHeightLimit = 0;
				exits[] = { {5,0,0}, {5,10,0} };
				loadingAngle = 60;
				loadingDistance = 5;
				maxLoadMass = 10000;
				unloadingInterval = 2;
				cargoSpacing[] = { 0, 0.15, 0 };
			};
		};
	};
	class Truck_02_base_F;
	class Truck_02_transport_base_F: Truck_02_base_F {
		class VehicleTransport
		{
			class Carrier
			{
				cargoAlignment[] = { "front", "center" };
				cargoBayDimensions[] = { {-0.30,1.15,-0.80}, {0.40,-3.65,0.35} };
				disableHeightLimit = 0;
				exits[] = { {5,0,0}, {5,10,0} };
				loadingAngle = 60;
				loadingDistance = 5;
				maxLoadMass	= 10000;
				unloadingInterval = 2;
				cargoSpacing[] = { 0, 0.15, 0 };
			};
		};
	};
	class Truck_03_base_F;
	class O_Truck_03_transport_F: Truck_03_base_F {
		class VehicleTransport
		{
			class Carrier
			{
				cargoAlignment[]= { "front", "center" };
				cargoBayDimensions[] = { {-0.35,0.05,-0.45}, {0.45,-4.95,0.65} };
				disableHeightLimit = 0;
				exits[] = { {5,0,0}, {5,10,0} };
				loadingAngle = 60;
				loadingDistance = 5;
				maxLoadMass	= 10000;
				unloadingInterval = 2;
				cargoSpacing[] = { 0, 0.15, 0 };
			};
		};
	};
	class O_Truck_03_covered_F: Truck_03_base_F {
		class VehicleTransport
		{
			class Carrier
			{
				cargoAlignment[] = { "front", "center" };
				cargoBayDimensions[] = { {-0.35,0.05,-0.45}, {0.45,-4.95,0.65} };
				disableHeightLimit	= 0;
				exits[] = { {5,0,0}, {5,10,0} };
				loadingAngle = 60;
				loadingDistance = 5;
				maxLoadMass = 10000;
				unloadingInterval = 2;
				cargoSpacing[] = { 0, 0.15, 0 };
			};
		};
	};
	class Truck_03_base_F;
	class O_Truck_03_transport_F: Truck_03_base_F {
		class VehicleTransport
		{
			class Carrier
			{
				cargoAlignment[] = { "front", "center" };
				cargoBayDimensions[] = { {-0.35,0.05,-0.45}, {0.45,-4.95,0.65} };
				disableHeightLimit	= 0;
				exits[] = { {5,0,0}, {5,10,0} };
				loadingAngle = 60;
				loadingDistance = 5;
				maxLoadMass = 10000;
				unloadingInterval = 2;
				cargoSpacing[] = { 0, 0.15, 0 };
			};
		};
	};
	class O_Truck_03_covered_F: Truck_03_base_F {
		class VehicleTransport
		{
			class Carrier
			{
				cargoAlignment[] = { "front", "center" };
				cargoBayDimensions[] = { {-0.35,0.05,-0.45}, {0.45,-4.95,0.65} };
				disableHeightLimit	= 0;
				exits[] = { {5,0,0}, {5,10,0} };
				loadingAngle = 60;
				loadingDistance = 5;
				maxLoadMass	= 10000;
				unloadingInterval = 2;
				cargoSpacing[] = { 0, 0.15, 0 };
			};
		};
	};
};
```
Items loaded into The Unarmed UGV will clip with the weapon station pylon, but I think that is acceptable. It will look like the top image. I made the dimensions so that everything up to a spare tire will fit.
The armed UGV has a smaller cargo space in front of the weapon station, it fits a smaller ammo or medical crate.

The trucks have the cargo space defined along the center line of the cargo area. larger crates might clip with soldiers' feet but they are centered so this will be kept to a minimum.

Truck_01 is the NATO HEMTT. The covered HEMTT and other factions' versions inherits from version defined above.
Truck_03 i CSAT Typhoon. The covered version does not inherit from this but it does have a base class which is also defined here.
Truck_02 is Kamaz, unfortunately the covered version inherits from Truck_02 base class and will need specific config entries for every faction, therefore only the open-top version is defined here.
